### ISSUE TITLE
Improvements to "Only Me" annotations

### ISSUE THREAD

**Issue by nickstenning** [hypothesis/product-backlog#110]
_This is a record of a feature request sent to our support address: [#836](https://hypothesis.zendesk.com/agent/tickets/836)._

> On a page, after you load the bookmarklet, you will see all the annotations from that page, filtered based on the selected group (and you can choose between 'Public' or any other group you are part of). Any plan to add a new option there? "Only mine" - to see only mine annotations from that page?

This is consistent with regular feedback that we get where people are frustrated that Only Me is inside of individual groups instead of a universal layer.  (Prior context: https://github.com/hypothesis/product-backlog/issues/200 and also https://github.com/hypothesis/product-backlog/issues/528)  This leads to usability issues, where they are confused about where their annotation is, and whether it's public or not, such as this issue: https://github.com/hypothesis/h/issues/5111

Should we allow you to set your view into a personal (Only Me) mode such that any *new* annotations you made would also be Only Me by default, and so that you could see all your Only Me annotations together without any others distracting you?  Should we also allow you to view your "Only Me" annotations from our search interface (activity pages) wherever they were made?  Since the group control is sticky (stays in the group you set it to, even across sites), you would be able to maintain that setting across sites, which has also been a request.

This is also related to the concept of an "All" scope or "group", where a user could see all annotations from any group on a page without having to click into each one.  (https://github.com/hypothesis/vision/issues/207)

**Comment by klemay** [comment]
Another user request: https://hypothesis.zendesk.com/agent/tickets/2683

**Comment by whoinoi** [comment]
Yes this feature is very much needed. I hope you guys will implement this. The issue has been open for 2 years already.

**Comment by judell** [comment]
You can do this by querying for your username in the sidebar:

![image](https://user-images.githubusercontent.com/46509/42703706-2a2a0112-8682-11e8-8da0-1b4d64208761.png)

A link to that view:  https://example.com/#annotations:query:user:judell

A tool for producing sets of such links for a group: https://jonudell.info/h/ContributorFocus/

**Comment by dwhly** [comment]
> You can do this by querying for your username in the sidebar

Though this only works for the current group.  Because we currently allow Only Me annotations *under every group* there could be Only Me annotations for the current page in a variety of places.

So:
1- We might break Only Me annotations out of groups into their own thing.
2- We should improve access to them overall.  

2 will most likely naturally fall from 1.

Another Q is:  "Should we allow another capability which is to draft an annotation and store it as a draft before making it visible to everyone in the current context."  Since Only Me annotations are currently used as a way to do this.  If we moved them to their own place, it would make the drafting workflow more challenging.

**Comment by klemay** [comment]
> Another Q is: "Should we allow another capability which is to draft an annotation and store it as a draft before making it visible to everyone in the current context." Since Only Me annotations are currently used as a way to do this. If we moved them to their own place, it would make the drafting workflow more challenging.

Or, if we allow for moving annotations among groups/layers (as in https://github.com/hypothesis/product-backlog/issues/111), users could draft annotations, set visibility to "Only Me," and then move to the appropriate group layer when ready.

**Comment by AdamDemirel** [comment]
Another problem with just querying your user is that although it filters the list, other's annotations/highlights are still visible on the page itself.

**Comment by dwhly** [comment]
> Another problem with just querying your user is that although it filters the list, other's annotations/highlights are still visible on the page itself.

People have suggested that this should be changed, and personally I agree.

**Comment by klemay** [comment]
More requests for this feature here: https://groups.google.com/forum/#!msg/hypothesis-forum/4kc4d8UXBk4/SdS0tjgxDAAJ

**Comment by klemay** [comment]
One user has [requested](https://groups.google.com/d/msg/hypothesis-forum/Sw6mB005gfU/il43yEXdDAAJ) the ability to make this a persistent / global configuration:

> the experience for use cases like mine would be greatly improved if eye icon state persisted across sites or at least the domain.

**Comment by thesofakillers** [comment]
Hi Guys, is this still in the works? I love hypothesis but often I find that I would much rather only see my annotations on a given page, and that annotations from other users can cause clutter.

**Comment by klemay** [comment]
Hi @thesofakillers, thanks for asking about this! We currently don't have plans to work on this, as we're primarily focused on our LMS app at the moment. 

I've added this issue to the agenda for our next backlog review meeting, scheduled to take place in three weeks. If this ends up getting placed into a higher priority position, I'll let you know.

**Comment by klemay** [comment]
Related to https://github.com/hypothesis/product-backlog/issues/111
